pandas/tests/io/json/test_compression.py

```diff
@@ -0,0 +1,156 @@
+import pytest
+import moto
+
+import pandas as pd
+from pandas import compat
+import pandas.util.testing as tm
+from pandas.util.testing import assert_frame_equal, assert_raises_regex
+
+
+COMPRESSION_TYPES = [None, 'bz2', 'gzip', 'xz']
+
+
+def test_compress_gzip():
```

Comment:
pls parametrize all of this

see how this is done in other compression tests

Reply:
I used the pattern that was used in the tests of compression with `pickle`.

Where there is a function which is used to decompress files by compression type.

Reply:
The zip tests will IMO always need to be special cases, as there isn't a writer we will always need to read from a fixture.